company_from_url/models/ir_rule.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 AppsToGROW - Henrik Norlin
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import time
+from collections import defaultdict
+
+from odoo import api, fields, models, tools, SUPERUSER_ID, _
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+from odoo.tools.safe_eval import safe_eval
+
+import logging
+_logger = logging.getLogger(__name__)
+
+class IrRule(models.Model):
+    _inherit = 'ir.rule'
+
+    ''' COMPUTE DOMAIN PER COMPANY'''
+
+    # /base/ir/ir_rule.py
+    # This did not work:
+    # @api.model
+    # @tools.ormcache('self.env.uid', 'self.env.user.company_id', 'model_name', 'mode') #UPDATED
+    # def _compute_domain(self, model_name, mode="read"):
+    #     return super(IrRule, self)._compute_domain(model_name, mode=mode)
+
+
+    # https://github.com/odoo/odoo/blob/10.0/odoo/addons/base/ir/ir_rule.py
+    # - @tools.ormcache('self._uid', 'model_name', 'mode')
+    # + @tools.ormcache('self.env.uid', 'self.env.context.get("company_id")', 'model_name', 'mode')
+    @api.model
+    @tools.ormcache('self.env.uid', 'self.env.user.company_id', 'model_name', 'mode')
```

Comment:
@norlinhenrik Instead of `self.env.context.get("company_id")` I made this with `self.env.user.company_id`, which also seems to work, and then I could comment out the context-related code in `res_users.py`. Do you think that will lead to problems?